### ISSUE TITLE
docs: Next Minimal Starter v10 Upgrade

### ISSUE THREAD

**Issue by ekqt** [trpc/trpc#2807]
### Area of Improvement

I would recommend matching the `next-minimal-starter` sample project with the [Usage with Next.js](https://trpc.io/docs/v10/nextjs) docs page. Here are a couple of changes that I would suggest:

* Usage with Next.js suggests [router merging](https://trpc.io/docs/v10/merging-routers), so we should provide the example with such and adopt the file structure recommended by [Recommended file structure](https://trpc.io/docs/v10/nextjs#recommended-file-structure).
* Introduce a second sub-router to fetch posts from an open API such as [jsonplaceholder](https://jsonplaceholder.typicode.com/) to illustrate an additional query and batching.
* Index page should list Discord server and GitHub discussions as other examples. 
* Update the link from `/example-apps`, [click here](https://github.com/trpc/trpc/tree/main/examples/minimal) to visit the href.

### Link to related docs

https://trpc.io/docs/v10/nextjs

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing the suggested changes!

**Comment by KATT** [comment]
The point of the example is that we want it to be minimal, but I do understand that we're breaking our own conventions here.


@trpc/trpc-core WDYT?

**Comment by bautistaaa** [comment]
I think its fine -as is- since its in fact a "minimal" example.

We have next-prisma-starter that addresses these as well.

> Index page should list Discord server and GitHub discussions as other `examples.
I think this is a good idea

> Introduce a second sub-router to fetch posts from an open API such as [jsonplaceholder](https://jsonplaceholder.typicode.com/) to illustrate an additional query and batching.
This is good feedback but i wouldnt put it in minimal(suggestions?)

**Comment by juliusmarminge** [comment]
I remember the initial point of the minimal was that someone felt overwhelmed by all the files so separating it I don't think is a good idea. This is, as the name suggests, the minimal setup required to get tRPC+Next working together.

Having it as 2 files, 1 server and 1 client, also helps for the QuickStart stackblitz embed on https://trpc.io#try-it-out.

> Index page should list Discord server and GitHub discussions as other `examples.

This we could absolutely do. Could make it a bit prettier too I guess as long as it isn't too distracting 🤔

**Comment by ekqt** [comment]
@KATT I didn't know it was the example used for the [https://trpc.io/?v=10#try-it-out](https://trpc.io/?v=10#try-it-out).

In that case, I'd argue that it would make a good additional example or to be integrated to another existing example. Here are the reasons for a starter project with this structure:

* I think that a basic example using only Next.js and tPRC (maybe call it serverless?) is very easy to understand and get it up and running without having to use other technologies such as Prisma et others.
* A starter project like this that has a server directory set with sub-routers gives you a clear idea on how it would scale and maintains consistency with the docs.
* Having a simple example as such could also be used with other technologies instead of Prisma such as Supabase (to name a db Baas) or Sanity (to name a headless CMS).

@bautistaaa @juliusmarminge  I'd be happy to pull together a working example with prototyping/testing API before committing to anything. If any other prototyping APIs would be better than [JSONplaceholder](https://jsonplaceholder.typicode.com/) I'd be happy to try them out. I think it would still be very minimal and easy to understand.

**Comment by ekqt** [comment]
Hello, again. I did end up making that example repo. I used [JSONplaceholder](https://jsonplaceholder.typicode.com/) to prototype how it'd work with a BaaS like Supabase/Firebase, headless-CMS like Sanity or any other third-party API.

Here's the repo: [tRPC-basic-starter](https://github.com/ekqt/trpc-basic-starter)
Here's the project demo: [tRPC Basic Starter](https://trpc-basic-starter.vercel.app/)

Lmk what you guys think. Cheers.
